### PR TITLE
Add example for writing a mask-supporting layer

### PR DESCRIPTION
**Pull request by mark86092**
As title, now we can have

1. The [doc](https://keras.io/layers/writing-your-own-keras-layers/) has the guide for writing custom layer, and the example `examples/antirectifier.py`.
2. This PR demonstrates mask supporting

### CODE REVIEW

**Comment by fchollet** [comment]
There are two things that one need to know about implementing mask-supporting layers:

If it's a layer that can *use* a mask (e.g. RNN):

- `self.supports_masking = True` should be set
- `call` should have a `mask` argument (with default None) and use it

If it's a layer that *creates* a mask (only `Masking` and `Embedding` do this, so it's not really something we need to talk about):

- should implement `compute_mask`

I don't really understand how the example shown conveys the information above. Maybe you could rewrite it along these lines?

**Comment by fchollet** [comment]
Closing this, please reopen with a new example if you would like to proceed
